### DISCUSSION TITLE
I discovered an error in the changes made in commit 03a6cbc5a5187e90f265b11841854afd9cfe7c54. The issue is that the scaling factor was not added to the symbol:

### DISCUSSION THREAD

**Discussion by goodgame365** [CoderLine/alphaTab#1753] [General]
I discovered an error in the changes made in commit 03a6cbc5a5187e90f265b11841854afd9cfe7c54. The issue is that the scaling factor was not added to the symbol:

at:\src\rendering\glyphs\SustainPedalGlyph.ts
----------------
`if (marker.pedalType === SustainPedalMarkerType.Down) {  
    canvas.fillMusicFontSymbol(markerX, y + h, 1, MusicFontSymbol.KeyboardPedalPed, true);  
    linePadding =  
        (SustainPedalGlyph.TextWidth / 2) * this.scale + SustainPedalGlyph.TextLinePadding * this.scale;  
} else if (marker.pedalType === SustainPedalMarkerType.Up) {  
    canvas.fillMusicFontSymbol(markerX, y + h, 1, MusicFontSymbol.KeyboardPedalUp, true);  `

In this part, the line:
canvas.fillMusicFontSymbol(markerX, y + h, 1, MusicFontSymbol.KeyboardPedalPed, true);  

should be changed to:
`canvas.fillMusicFontSymbol(markerX, y + h, this.scale, MusicFontSymbol.KeyboardPedalPed, true);  `

Similarly, the latter part should also change 1 to this.scale.

**Comment by goodgame365** [comment]
Additionally, I noticed that there is only a visual effect, but no sustain effect during playback. I'm not sure if this is the current situation.

**Comment by Danielku15** [comment]
> I discovered an error in the changes made in commit https://github.com/CoderLine/alphaTab/commit/03a6cbc5a5187e90f265b11841854afd9cfe7c54. The issue is that the scaling factor was not added to the symbol:

`1` is as expected and correct. The scale there is an additional relative scale factor of the glyph. Music symbols are drawn with SmuFL fonts at a pre-configured font size. This scale factor is additionally applied: 

https://github.com/CoderLine/alphaTab/blob/develop/src/platform/svg/CssFontSvgCanvas.ts#L51
https://github.com/CoderLine/alphaTab/blob/develop/src/platform/javascript/Html5Canvas.ts#L283

Why do you think it is wrong? Did you notice any wrong rendering?

> Additionally, I noticed that there is only a visual effect, but no sustain effect during playback. I'm not sure if this is the current situation.

Yes, for now this is on purpose.

**Comment by goodgame365** [comment]
@Danielku15 

The situation is that, for the scaling ratio, when scaled to 0.5, this symbol does not shrink accordingly, as shown in the figure.
1、 This is the state without scaling：
![1](https://github.com/user-attachments/assets/199e5087-f1b3-441f-b6d8-cd99d3a9d886)

2、This is the state scaled down by 50%.:
![2](https://github.com/user-attachments/assets/0ab7f735-a5a9-4b88-80ea-fe998f501d26)

I think this is the correct one”或“I believe this is the right one. like this img
![normal](https://github.com/user-attachments/assets/e893e5a3-f44f-4e73-80e7-3fa482902e9d)
